yt_dlp/extractor/mediaset.py

```diff
@@ -129,71 +85,23 @@ class MediasetIE(ThePlatformBaseIE):
-    }, {
-        # iframe twitter (from http://www.wittytv.it/se-prima-mi-fidavo-zero/)
-        'url': 'https://static3.mediasetplay.mediaset.it/player/index.html?appKey=5ad3966b1de1c4000d5cec48&programGuid=FAFU000000665104&id=665104',
-        'only_matching': True,
-    }, {
-        # embedUrl (from https://www.wittytv.it/amici/est-ce-que-tu-maimes-gabriele-5-dicembre-copia/)
-        'url': 'https://static3.mediasetplay.mediaset.it/player/v2/index.html?partnerId=wittytv&configId=&programGuid=FD00000000153323&autoplay=true&purl=http://www.wittytv.it/amici/est-ce-que-tu-maimes-gabriele-5-dicembre-copia/',
-        'only_matching': True,
     }, {
         'url': 'mediaset:FAFU000000665924',
         'only_matching': True,
-    }, {
-        'url': 'https://www.mediasetplay.mediaset.it/video/mediasethaacuoreilfuturo/palmieri-alicudi-lisola-dei-tre-bambini-felici--un-decreto-per-alicudi-e-tutte-le-microscuole_FD00000000102295',
-        'only_matching': True,
-    }, {
-        'url': 'https://www.mediasetplay.mediaset.it/video/cherryseason/anticipazioni-degli-episodi-del-23-ottobre_F306837101005C02',
-        'only_matching': True,
-    }, {
-        'url': 'https://www.mediasetplay.mediaset.it/video/tg5/ambiente-onda-umana-per-salvare-il-pianeta_F309453601079D01',
-        'only_matching': True,
-    }, {
-        'url': 'https://www.mediasetplay.mediaset.it/video/grandefratellovip/benedetta-una-doccia-gelata_F309344401044C135',
-        'only_matching': True,
-    }, {
-        'url': 'https://www.mediasetplay.mediaset.it/movie/herculeslaleggendahainizio/hercules-la-leggenda-ha-inizio_F305927501000102',
-        'only_matching': True,
-    }, {
-        'url': 'https://mediasetinfinity.mediaset.it/video/braveandbeautiful/episodio-113_F310948005000402',
-        'only_matching': True,
-    }, {
-        'url': 'https://static3.mediasetplay.mediaset.it/player/v2/index.html?partnerId=wittytv&configId=&programGuid=FD00000000153323',
-        'only_matching': True,
     }]
 
-    def _extract_from_webpage(self, url, webpage):
-        def _program_guid(qs):
-            return qs.get('programGuid', [None])[0]
-
-        for mobj in re.finditer(
-                r'<iframe\b[^>]+\bsrc=(["\'])(?P<url>(?:https?:)?//(?:www\.)?video\.mediaset\.it/player/playerIFrame(?:Twitter)?\.shtml.*?)\1',
-                webpage):
-            embed_url = mobj.group('url')
-            embed_qs = parse_qs(embed_url)
-            program_guid = _program_guid(embed_qs)
-            if program_guid:
-                yield self.url_result(embed_url)
-                continue
-
-            video_id = embed_qs.get('id', [None])[0]
-            if not video_id:
-                continue
-            urlh = self._request_webpage(embed_url, video_id, note='Following embed URL redirect')
-            embed_url = urlh.geturl()
-            program_guid = _program_guid(parse_qs(embed_url))
-            if program_guid:
-                yield self.url_result(embed_url)
-
```

Comment:
Used by GenericIE for embed detection

https://github.com/yt-dlp/yt-dlp/blob/88f20eff1b13f70d332ef97fc1be175271bec130/yt_dlp/extractor/generic.py#L1550-L1562

Reply:
Since we are cleaning up this extractor, could u also move the above test into `_WEBPAGE_TESTS` under the IE

Reply:
added `_EMBED_REGEX` and moved `_WEBPAGE_TESTS` from generic to the mediaset extractor.

How do I test this?

Reply:
`python3 test/test_download.py TestDownload.test_Mediaset_webpage`
`python3 test/test_download.py TestDownload.test_Mediaset_webpage_1` etc

Reply:
yes, thanks. I figured it out by searching in the code how the tests are generated ;-)